src/pages/NewChatPage.js

```diff
@@ -194,8 +201,15 @@ class NewChatPage extends Component {
         if (userLogins.length < 1) {
             return;
         }
-
-        Report.fetchOrCreateChatReport([this.props.session.email, ...userLogins]);
+        const formattedUserLogins = _.map(userLogins, login => OptionsListUtils.addSMSDomainIfPhoneNumber(login));
+        let newChat = {};
+        const chat = ReportUtils.getChatByParticipants(formattedUserLogins, this.props.reports);
+        if (!chat) {
+            newChat = ReportUtils.buildOptimisticChatReport(formattedUserLogins);
+        }
+        const reportID = chat ? chat.reportID : newChat.reportID;
+        Report.openReport(reportID, newChat.participants, newChat);
+        Navigation.navigate(ROUTES.getReportRoute(reportID));
```

Comment:
NAB: The same code is duplicated above so it might be nice to extract this to a function.

Reply:
+1